azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -121,122 +128,107 @@ private FlowContainer(final Path projectDirPath, final Props props)
     } catch (NumberFormatException ne) {
-      System.out.printf("Execution ID %s is invalid %n", args[0]);
+      logger.error("Execution ID passed in argument is invalid {}", execIdStr);
+      throw new ExecutorManagerException(ne);
     }
-    System.out.printf("Execution ID = %d%n", execId);
 
-    // Setup work directories
-    final Path currentWorkingDir = Paths.get("").toAbsolutePath();
-    final Path projectDirPath = Paths.get(currentWorkingDir.toString(), PROJECT_DIR);
-    final Path jobtypePluginPath = Paths.get(currentWorkingDir.toString(), JOBTYPE_DIR);
+    logger.info("Execution ID : " + execId);
 
-    // Move files to respective dirs
-    // Create project dir
-    // TODO: Need to revisit
-    System.out.println("Creating project dir");
-    Files.createDirectory(projectDirPath);
-    Path projectZipPath = Paths.get(projectDirPath.toString(), projectZipName);
-    System.out.println("moving projectDir:" + projectZipName + ": to " +
-        projectZipPath);
-    Files.move(Paths.get(projectZipName), projectZipPath);
-    // TODO : revisit this logic,
-    // Unzip the project zip
-    FlowContainer.unzipFile(projectZipPath.toString(), projectDirPath);
+    // AZ_HOME must provide a correct path, if not then azHome is set to current working dir.
+    final String azHome = Optional.ofNullable(System.getenv(AZ_HOME)).orElse("");
 
-    // TODO : Deepak - throw away after directory setup is final.
-    Path pwd = Paths.get("").toAbsolutePath();
-    System.out.println("Deepak : Dumping dir structure for = " + pwd.toString());
-    java.nio.file.Files.walk(pwd).filter(java.nio.file.Files::isRegularFile)
-        .forEach(System.out::println);
+    final Path currentDir = Paths.get(System.getenv(AZ_HOME)).toAbsolutePath();
 
     // Set Azkaban props
-    Props props = setAzkabanProps(jobtypePluginPath);
+    final Path jobtypePluginPath = Paths.get(currentDir.toString(), JOBTYPE_DIR);
+    Props azkabanProps = FlowContainer.setAzkabanProps(jobtypePluginPath);
 
-    // Inject AzkabanCommonModule
-    final Injector injector = Guice.createInjector(
-        new AzkabanCommonModule(props)
-    );
-    SERVICE_PROVIDER.setInjector(injector);
+    // Setup Injector
+    FlowContainer.setInjector(azkabanProps);
+
+    // Setup work directories
+    final String execDirPath = FlowContainer.setupWorkDir(currentDir, projectZipName);
 
     // Constructor
     final FlowContainer flowContainer =
-        new FlowContainer(projectDirPath, props);
+            new FlowContainer(azkabanProps, execDirPath);
 
-    // Use some execId to execute the flow
+    // TODO : Revisit this logic with full implementation for JMXBEanManager and other callback mechanisms
+    JmxJobMBeanManager.getInstance().initialize(azkabanProps);
+    // execute the flow
     flowContainer.submitFlow(execId);
   }
 
-  // Set Azkaban Props
+  private static String setupWorkDir(final Path currentDir, final String projectZipName)
```

Comment:
I think this should really be part of PREPARING stage. Calling it over here makes it part of DISPATCHING state. Is there a way this can be moved inside createFlowRunner() after the state is moved to PREPARING?